sources/router.c

```diff
@@ -144,6 +175,13 @@ int od_router_reconfigure(od_router_t *router, od_rules_t *rules)
 			od_rule_key_free(rk);
 		}
 
+		od_list_foreach(&to_drop, i)
```

Comment:
`od_list_foreach_safe` and above code should also use this MARCO, no idea why is this working without sigsegv for so long